library/src/scala/tasty/reflect/Printers.scala

```diff
@@ -847,15 +837,80 @@ trait Printers
 
       }
 
+      def flatBlock(stats: List[Statement], expr: Term): (List[Statement], Term) = {
+        val flatStats = List.newBuilder[Statement]
+        def extractFlatStats(stat: Statement): Unit = stat match {
+          case Term.Block(stats1, expr1) =>
+            val it = stats1.iterator
+            while (it.hasNext)
+              extractFlatStats(it.next())
+            extractFlatStats(expr1)
+          case Term.Inlined(_, bindings, expansion) =>
+            val it = bindings.iterator
+            while (it.hasNext)
+              extractFlatStats(it.next())
+            extractFlatStats(expansion)
+          case Term.Literal(Constant.Unit()) => // ignore
+          case stat => flatStats += stat
+        }
+        def extractFlatExpr(term: Term): Term = term match {
+          case Term.Block(stats1, expr1) =>
+            val it = stats1.iterator
+            while (it.hasNext)
+              extractFlatStats(it.next())
+            extractFlatExpr(expr1)
+          case Term.Inlined(_, bindings, expansion) =>
+            val it = bindings.iterator
+            while (it.hasNext)
+              extractFlatStats(it.next())
+            extractFlatExpr(expansion)
+          case term => term
+        }
+        val it = stats.iterator
+        while (it.hasNext)
+          extractFlatStats(it.next())
+        val flatExpr = extractFlatExpr(expr)
+        (flatStats.result(), flatExpr)
+      }
+
+      def printFlatBlock(stats: List[Statement], expr: Term): Buffer = {
+        val (stats1, expr1) = flatBlock(stats, expr)
+        // Remove Term.Lambda nodes, lambdas are printed by their definition
+        val stats2 = stats1.filter { case Term.Lambda(_, _) => false; case _ => true }
+        val (stats3, expr3) = expr1 match {
+          case Term.Lambda(_, _) =>
+            val init :+ last  = stats2
+            (init, last)
+          case _ => (stats2, expr1)
+        }
```

Comment:
The `Remove Term.Lambda` logic seems to belong to somewhere else .

Reply:
For the decompiler, `Term.Lambda` are `no-op`/`no-print` trees that must be removed from the blocks. We need to remove them here because they might affect the block flattening.